src/main/java/service/StudentService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import entity.Student;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class StudentService {
+
+
+    public Double averageRatingForSubject(List<Student> students, String subjectName){
+        return students.stream().map(Student::getRating).mapToDouble(d -> d.get(subjectName))
+                .average().orElseThrow();
+    }
+
+    public void printStudentsAndSubjects(List<Student> students){
+        List<String> subjects = new LinkedList<>();
+
+
+        students.stream().map(Student::getRating)
```

Comment:
I guess you can somehow refactor this logic to get subjects without redundant initialization of these subjects